### PR TITLE
Feature/rebase config via commandline

### PR DESCRIPTION
**Pull request by dobrusev**
This feature enables command line config instead of requirement to have .munit and target.hxml files available. Some undocumented command line arguments has been used from original approach `munit run {config} {src} {testbin} {report}` the rest is avaialbe using **config:** or **target:** prefixes. Following .munit:

```
version=munit
src=src/test/haxe
bin=bin/test
report=bin/test-report
hxml=munit.hxml
classPaths=src/main/haxe
coverageIgnoredClasses=module
```

and target.hxml file:

```
#neko
-main TestMain
-lib munit
-cp src/main/haxe
-cp src/test/haxe
-neko bin/test/neko_test.n
```

can be now executed as following command (notice **cliconfig**):

```
haxelib run munit test cliconfig src/test/haxe bin/test bin/test-report config:classPath src/main/haxe config:coverageIgnoredClass module target:-main TestMain target:-lib munit target:-cp src/main/haxe target:-cp src/test/haxe target:-neko bin/test/neko_test.n
```

Updates in code should not affect any previous munit functionality. If suggested changes are not welcomed let me know, I will make a repo fork instead...

### CODE REVIEW

File: test/TestSuite.hx [hunk 1]
```diff
@@ -1,20 +1,20 @@
 import massive.munit.TestSuite;
 
-import massive.munit.AssertionExceptionTest;
 import massive.munit.AssertTest;
-import massive.munit.async.AsyncDelegateTest;
+import massive.munit.TestSuiteTest;
```
Comment (jozefchutka) [inline test/TestSuite.hx]: sort please

File: test/TestSuite.hx [hunk 2]
```diff
@@ -26,20 +26,20 @@ class TestSuite extends massive.munit.TestSuite
 	{
 		super();
 
-		add(massive.munit.AssertionExceptionTest);
 		add(massive.munit.AssertTest);
-		add(massive.munit.async.AsyncDelegateTest);
+		add(massive.munit.TestSuiteTest);
```
Comment (jozefchutka) [inline test/TestSuite.hx]: revert this block, its not supposed to be changed

File: tool/src/massive/munit/command/RunCommand.hx [hunk 3]
```diff
@@ -26,15 +26,17 @@
...
 import haxe.io.Eof;
 import massive.haxe.log.Log;
 import massive.haxe.util.RegExpUtil;
-import massive.munit.ServerMain;
 import massive.munit.client.HTTPClient;
+import massive.munit.ServerMain;
 import massive.munit.util.MathUtil;
 import massive.sys.io.File;
 import massive.sys.io.FileSys;
+import neko.vm.Module;
```
Comment (jozefchutka) [inline tool/src/massive/munit/command/RunCommand.hx]: remove unused

Comment (dobrusev) [reply tool/src/massive/munit/command/RunCommand.hx]: this one is used line 275 -> Module.local().name

File: test/TestSuite.hx (hunk 2 above)
Comment (dobrusev) [reply test/TestSuite.hx]: I didn't manually changed it it's after I run the build.sh

File: test/TestSuite.hx (hunk 1 above)
Comment (dobrusev) [reply test/TestSuite.hx]: I didn't manually changed it it's after I run the build.sh

File: tool/src/massive/munit/Config.hx [hunk 4]
```diff
@@ -54,9 +57,107 @@ class Config
...
 		this.currentVersion = currentVersion;
 		targetTypes = defaultTargetTypes;
-		configFile = dir.resolveFile(".munit");
-		exists = configFile.exists;
-		if(exists) load();
+		if(!parseCommandLineConfig())
+		{
+			configFile = dir.resolveFile(".munit");
+
+			exists = configFile.exists;
```
Comment (elsassph) [inline tool/src/massive/munit/Config.hx]: Why the reformatting?
```
configFile = dir.resolveFile(".munit");
exists = configFile.exists;
if(exists) load();
```

File: tool/src/massive/munit/command/RunCommand.hx [hunk 5]
```diff
@@ -267,7 +269,11 @@ class RunCommand extends MUnitTargetCommandBase
 			//Windows has issue releasing port registries reliably.
 			//To prevent possibility of nekotools server failing, on
 			//windows the tmp directory is always located inside the munit install
-			FileSys.setCwd(console.originalDir.nativePath);
+			//FileSys.setCwd(console.originalDir.nativePath);
```
Comment (elsassph) [inline tool/src/massive/munit/command/RunCommand.hx]: Remove commented original line

File: tool/src/massive/munit/Config.hx [hunk 6]
```diff
@@ -54,9 +57,103 @@ class Config
...
+	}
+
+	private function getCommandLineConfigValues(name:String):Array<String>
+	{
+		var args = Sys.args();
+		var result:Array<String> = [];
+		var length = args.length;
+		for(i in 0...length)
+		{
+			if(args[i] == ("config:" + name) && length > i)
```
Comment (SlavaRa) [inline tool/src/massive/munit/Config.hx]: `length > i` is always `true`

File: tool/src/massive/munit/Config.hx [hunk 7]
```diff
@@ -54,9 +57,103 @@ class Config
...
+		var hxml = "";
+		var flag = "target:";
+		var args = Sys.args();
+		var type:TargetType = null;
+		var file:File = null;
+		var length = args.length;
+		for(i in 0...length)
+		{
+			var arg = args[i];
+			if(StringTools.startsWith(arg, flag) && length > i)
```
Comment (SlavaRa) [inline tool/src/massive/munit/Config.hx]: `length > i` is always `true`

File: tool/src/massive/munit/Config.hx (hunk 6 above)
Comment (jozefchutka) [reply tool/src/massive/munit/Config.hx]: @dobrusev this we need to fix to `length > (i + 1)` or `(i + 1) < length` so we can access `args[i + 1]`

Comment (SlavaRa) [reply tool/src/massive/munit/Config.hx]: or `for(i in 0...length-1)`

Comment (jozefchutka) [reply tool/src/massive/munit/Config.hx]: yep, even better

**Pull request merged**